pkg/volume/iscsi/iscsi_util.go

```diff
@@ -80,22 +90,27 @@ func (util *ISCSIUtil) MakeGlobalPDName(iscsi iscsiDisk) string {
 }
 
 func (util *ISCSIUtil) AttachDisk(b iscsiDiskBuilder) error {
-	devicePath := strings.Join([]string{"/dev/disk/by-path/ip", b.portal, "iscsi", b.iqn, "lun", b.lun}, "-")
-	exist := waitForPathToExist(devicePath, 1)
+	var devicePath string
+	if b.iface == "default" {
+		devicePath = strings.Join([]string{"/dev/disk/by-path/ip", b.portal, "iscsi", b.iqn, "lun", b.lun}, "-")
+	} else {
+		devicePath = strings.Join([]string{"/dev/disk/by-path/pci", "*", "ip", b.portal, "iscsi", b.iqn, "lun", b.lun}, "-")
```

Comment:
I think [this](https://github.com/systemd/systemd/blob/master/src/udev/udev-builtin-path_id.c#L665) is what you want.

You can take a look at the openstack changes that do the same [here](https://github.com/openstack/os-brick/commit/15a55bd5d1e3ad274afafb43b96b8b88bb7d6ff8) and [here](https://github.com/openstack/nova/commit/554647a4deee6ece221eb79fc93551de72b17ae3)